content/react/en/creating-addons.md

```diff
@@ -0,0 +1,429 @@
+---
+title: "Creating addons"
+tocTitle: "Creating addons"
+description: "Learn how to build your own addons that will super charge your development"
+commit: "dac373a"
+---
+
+# Creating addons
+
+Storybook's addon architecture allows you to expand it's feature set and customize it to your needs.
+
+Writing your own addon may sound a bit daunting, but as you'll see in this guide, it can be quite simple.
+
+We're going to write an addon together, first let's scope what our addon will do:
+
+## The addon we're going to write
+
+Your team has design assets which have some relationship to the UI components. That relationship isn't really apparent from within storybook's UI. Let's fix that!
+
+We're going to write an addon that will show our design assets next to our component!
+
+Our addon should:
+- display the design asset in a panel
+- support images, but also urls for embedding
+- should support multiple assets, just in case there will be multiple versions or themes
+
+The way we'll be attaching the list of assets to stories is via parameters.
+
+This is what it will look like when someone is adding design assets to their story:
+
+```js
+import { storiesOf } from '@storybook/react';
+import Component from './component';
+
+storiesOf('Component')
```

Comment:
Could we refactor this guide to use the new CSF?

Reply:
✅